### PR TITLE
Add Trivy scan for release branches

### PR DESCRIPTION
**Pull request by maci3jka**
This pr adds trivy scan for release branches
KU-2031

*Also verify you have:*

* [x] Read the [contributions](https://github.com/ubuntu/microk8s/blob/master/CONTRIBUTING.md) page.
* [x] Submitted the [CLA form](https://ubuntu.com/legal/contributors/agreement), if you are a first time contributor.

### CODE REVIEW

File: .github/workflows/trivy.yaml [hunk 1]
```diff
@@ -6,18 +6,35 @@ on:
...
+    steps:
+      - name: Checking out repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Create branches to test
+        id: branches
+        run: |
+          BRANCHES=$(git branch -r | grep  -E '^  origin\/(((v|release-)?[0-9]+.[0-9]+)|(master|main))$' | \
+            sed -e 's#^  origin/##'  | jq -R -s -c 'split("\n")[:-1]')
```
Comment (eaudetcobello) [inline .github/workflows/trivy.yaml:22]: Can you write one or two example strings that pass validation with this regex?

Comment (maci3jka) [reply .github/workflows/trivy.yaml:22]: sure, for this repo they are
```
  origin/1.28
  origin/master
  origin/release-1.30
  origin/release-1.31
  origin/v1.1
```

Comment (maci3jka) [reply .github/workflows/trivy.yaml:28]: @eaudetcobello fixed

File: .github/workflows/trivy.yaml [hunk 2]
```diff
@@ -6,18 +6,41 @@ on:
     - cron: '0 10 * * *'
 
 jobs:
+  prepare:
```
Comment (eaudetcobello) [inline .github/workflows/trivy.yaml:9]:
```suggestion
  prepare-branches-to-test:
```

File: .github/workflows/trivy.yaml [hunk 3]
```diff
@@ -6,18 +6,41 @@ on:
...
+          #  origin/v1.1
+          #  origin/release-1.30
+          #  origin/master
+          #  origin/main
+          BRANCHES=$(git branch -r | grep  -E '^  origin\/(((v|release-)?[0-9]+.[0-9]+)|(master|main))$' | \
+            sed -e 's#^  origin/##'  | jq -R -s -c 'split("\n")[:-1]')
+          echo "branches=$(echo $BRANCHES)" >> $GITHUB_OUTPUT
   scan:
     runs-on: ubuntu-latest
+    needs: prepare
```
Comment (eaudetcobello) [inline .github/workflows/trivy.yaml:32]:
```suggestion
    needs: prepare-branches-to-test
```

**Review by eaudetcobello — APPROVED** [review]
This is good to merge once you reviewed my suggestions

File: .github/workflows/trivy.yaml (hunk 3 above)
Comment (maci3jka) [reply .github/workflows/trivy.yaml:32]: how about prepare-branches-to-test?

File: .github/workflows/trivy.yaml [hunk 4]
```diff
@@ -6,18 +6,41 @@ on:
...
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      branches: ${{ steps.branches.outputs.branches }}
+    steps:
+      - name: Checking out repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Create branches to test
```
[the remark this answers is not in the document]
Comment (maci3jka) [reply .github/workflows/trivy.yaml:18]: `List branches to test` ?

**Pull request merged**